### DISCUSSION TITLE
Add GEX data without normalizing

### DISCUSSION THREAD

**Discussion by mitchTwoTimes** [GreenleafLab/ArchR#1418] [Questions / Documentation]
Is it possible to `addGeneExpressionMatrix` without normalizing? I was wondering if I could maintain the raw counts within ArchR and save normalization for when I export to Seurat. I tried passing `NULL` to `scaleTo` and received the following
```
Error in data.frame(scaleTo = scaleTo, exclude = excludeChr, stringsAsFactors = FALSE) : 
  arguments imply differing number of rows: 0, 2
```

Sorry I can't provide more - my RStudio session crashes before I can get the full `traceback()`.

**Comment by rcorces** [comment]
This isnt currently possible. And it would probably muck around with some of the base assumptions of other ArchR functions which assume that the RNA data is normalized. In `release_1.0.2`, `scaleTo` is not allowed to be `NULL`:
https://github.com/GreenleafLab/ArchR/blob/9b3e72e1fa82b9f6c116023e3ac09165d44b04a4/R/MatrixGeneExpression.R#L44

But that `.validInput()` call was added relatively recently. I think it should be relatively simple to get the un-normalized counts back out by reversing the math done here for normalization:
https://github.com/GreenleafLab/ArchR/blob/968e4421ce7187a8ac7ea1cf6077412126876d5f/R/HiddenUtils.R#L178
But I'm not sure that helps you very much.

**Reply by mitchTwoTimes** [reply]
Ok thanks for sharing that hack. Is there no log transformation that occurs?

**Reply by rcorces** [reply]
yes sorry there is also a log2(count+1) transform that occurs here:
https://github.com/GreenleafLab/ArchR/blob/968e4421ce7187a8ac7ea1cf6077412126876d5f/R/ArrowWrite.R#L206
